### ISSUE TITLE
Fatal error: Uncaught exception 'Alchemy\Zippy\Exception\RuntimeException'

### ISSUE THREAD

**Issue by ohadschn** [benbalter/wordpress-to-jekyll-exporter#96]
I'm getting the following error trying to export my blog:

```
Warning: fopen(D:\local\Temp/zippy_5998b60f385e4\./): failed to open stream: No such file or directory in D:\home\site\wwwroot\wp-content\plugins\jekyll-exporter\vendor\alchemy\zippy\src\Resource\Writer\StreamWriter.php on line 21 
Warning: stream_copy_to_stream() expects parameter 1 to be resource, boolean given in D:\home\site\wwwroot\wp-content\plugins\jekyll-exporter\vendor\alchemy\zippy\src\Resource\Writer\StreamWriter.php on line 24 
Warning: fclose() expects parameter 1 to be resource, boolean given in D:\home\site\wwwroot\wp-content\plugins\jekyll-exporter\vendor\alchemy\zippy\src\Resource\Writer\StreamWriter.php on line 25 
Fatal error: Uncaught exception 'Alchemy\Zippy\Exception\RuntimeException' with message 'unable to add ./\ to the zip file' in D:\home\site\wwwroot\wp-content\plugins\jekyll-exporter\vendor\alchemy\zippy\src\Adapter\ZipExtensionAdapter.php:346 Stack trace: #0 D:\home\site\wwwroot\wp-content\plugins\jekyll-exporter\vendor\alchemy\zippy\src\Adapter\ZipExtensionAdapter.php(285): Alchemy\Zippy\Adapter\ZipExtensionAdapter->addEmptyDir(Object(ZipArchive), './\\') #1 D:\home\site\wwwroot\wp-content\plugins\jekyll-exporter\vendor\alchemy\zippy\src\Adapter\ZipExtensionAdapter.php(207): Alchemy\Zippy\Adapter\ZipExtensionAdapter->addEntries(Object(Alchemy\Zippy\Adapter\Resource\ZipArchiveResource), Array, true) #2 D:\home\site\wwwroot\wp-content\plugins\jekyll-exporter\vendor\alchemy\zippy\src\Adapter\AbstractAdapter.php(58): Alchemy\Zippy\Adapter\ZipExtensionAdapter->doCreate('D:\\local\\Temp/w...', Array, true) #3 D:\home\site\wwwroot\wp-content\plugins\jekyll-exporter\vendor\alchemy\zippy\src\Zippy.php(68): Alchemy\Zippy\Adapter\ in D:\home\site\wwwroot\wp-content\plugins\jekyll-exporter\vendor\alchemy\zippy\src\Zippy.php on line 70 
```

### Environment
Azure Web App (IIS on Windows)
PHP 5.6
[Project Nami](http://projectnami.org/) (WordPress on MS SQL Server) 4.8.1 
Twenty Sixteen Theme 1.3
Site: https://www.ohadsoft.com

**Comment by ohadschn** [comment]
I believe this was caused due to Zippy resolving the Azure Web App sandbox default temp folder, [which has some weird behavior going on](
https://github.com/projectkudu/kudu/wiki/Understanding-the-Azure-App-Service-file-system#temporary-files).

I was able to resolve this issue by getting rid of Zippy (in [jekyll-exporter.php](https://github.com/benbalter/wordpress-to-jekyll-exporter/blob/6e36c09fab4356a64971dfe71fbc6c148588619a/jekyll-exporter.php#L418)):

```php
function addFolderToZip($source, $destination, $flags = ZIPARCHIVE::OVERWRITE)
{
	$source = realpath($source);
	$destination = realpath($destination);
			
	if (!file_exists($source)) {
		die("file does not exist: " . $source);
	}

	$zip = new ZipArchive();
	if (!$zip->open($destination, $flags )) {
		die("Cannot open zip archive: " . $destination);
	}
	
	$files = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($source), RecursiveIteratorIterator::SELF_FIRST);

	foreach ($files as $file)
	{
		// Ignore "." and ".." folders
		if( in_array(substr($file, strrpos($file, DIRECTORY_SEPARATOR)+1), array('.', '..')) )
			continue;
		
		if (is_dir($file) === true)
		{
			$zip->addEmptyDir(str_replace($source . DIRECTORY_SEPARATOR, '', $file . DIRECTORY_SEPARATOR));
		}
		else if (is_file($file) === true)
		{
			$zip->addFile($file, str_replace($source . DIRECTORY_SEPARATOR, '', $file));
		}
	}

	return $zip->close();
}


/**
 * Zip temp dir
 */
function zip() {

	$this->addFolderToZip($this->dir, $this->zip);
}
```

It's also one dependency less, so a win win in my book.

**Comment by benbalter** [comment]
@ohadschn if Zippy is not required, would you be able to submit a PR with the above code? It'd be great to be able to remove a dependency if we can!

**Comment by benbalter** [comment]
> I was able to resolve this issue by getting rid of Zippy altogether... It's also one dependency less, so a win win in my book.

Nice. Implemened via https://github.com/benbalter/wordpress-to-jekyll-exporter/commit/478c68bb9b9928e64603c1337a655eba3c8108e0

**Comment by ohadschn** [comment]
Thanks!

I noticed you removed the `realpath` normalizations at the top of `zip_folder`. I totally agree about `destination` - not only is it meaningless (it's really `ZipArchive`'s problem), it is actually a bug since the zip file typically doesn't exist at that point, and `realpath` fails for non-existent files (a fact I didn't previously know, and bit us in the other issue as well).

However, I believe it is best to keep `$source = realpath($source);` (probably after the existence test). Otherwise the following `substr` operations might use the wrong length (for example if `source` is something like `/foo/../bar`):
https://github.com/benbalter/wordpress-to-jekyll-exporter/blob/59b411423c6d4fae74fbc1f7c5d48f5bc164f170/jekyll-exporter.php#L446
https://github.com/benbalter/wordpress-to-jekyll-exporter/blob/59b411423c6d4fae74fbc1f7c5d48f5bc164f170/jekyll-exporter.php#L448

**Comment by benbalter** [comment]
Implemented via d3fa27f.
